Source: Based on Leonardo Boshell's changes in Gentoo
Upstream: No
Reason: Adds $ROOT support.
--- sgml-common-0.6.3/bin/install-catalog.in
+++ sgml-common-0.6.3/bin/install-catalog.in
@@ -63,62 +63,62 @@
 esac
 
 # Check that the super catalog can be created and changed and deleted
-if [ ! -w /etc/sgml ]
+if [ ! -w "$ROOT/etc/sgml" ]
 then
-  echo "`basename $0`: unable to write in /etc/sgml." >&2
+  echo "`basename $0`: unable to write in $ROOT/etc/sgml." >&2
   exit 2
 fi
 case $SGML_ACTION in
    addition)
-	if [ -e /etc/sgml/catalog -a ! -w /etc/sgml/catalog ]
+	if [ -e "$ROOT/etc/sgml/catalog" -a ! -w "$ROOT/etc/sgml/catalog" ]
 	then
-	  echo "`basename $0`: can not modify \"/etc/sgml/catalog\"." >&2
+	  echo "`basename $0`: can not modify \"$ROOT/etc/sgml/catalog\"." >&2
 	  exit 2
 	fi
 	;;
    removal)
-	if [ ! -w /etc/sgml/catalog ]
+	if [ ! -w "$ROOT/etc/sgml/catalog" ]
 	then
-	  echo "`basename $0`: can not modify \"/etc/sgml/catalog\"." >&2
+	  echo "`basename $0`: can not modify \"$ROOT/etc/sgml/catalog\"." >&2
 	  exit 2
 	fi
 	;;
 esac
 
 # Check that the centralized catalog can be created, changed and deleted
-if [ -z "$SGML_CENTRALIZED" ]
+if [ -z "$ROOT/$SGML_CENTRALIZED" ]
 then
   echo -e $SGML_HELP_MESSAGE >&2
   exit 1
 fi
 case $SGML_ACTION in
    addition)
-	if [ -e $SGML_CENTRALIZED -a ! -w $SGML_CENTRALIZED ]
+	if [ -e "$ROOT/$SGML_CENTRALIZED" -a ! -w "$ROOT/$SGML_CENTRALIZED" ]
 	then
-	  echo "`basename $0`: can not modify \"$SGML_CENTRALIZED\"." >&2
+	  echo "`basename $0`: can not modify \"$ROOT/$SGML_CENTRALIZED\"." >&2
 	  exit 2
 	fi
 	;;
    removal)
-	if [ ! -w $SGML_CENTRALIZED ]
+	if [ ! -w "$ROOT/$SGML_CENTRALIZED" ]
 	then
-	  echo "`basename $0`: can not modify \"$SGML_CENTRALIZED\"." >&2
+	  echo "`basename $0`: can not modify \"$ROOT/$SGML_CENTRALIZED\"." >&2
 	  exit 2
 	fi
 	;;
 esac
 
 # Check that we have at least one ordinary package to process
-if [ -z "$SGML_ORDINARY" ]
+if [ -z "$ROOT/$SGML_ORDINARY" ]
 then
   echo -e $SGML_HELP_MESSAGE >&2
   exit 1
 fi
 case $SGML_ACTION in
    addition)
-	if [ ! -s $SGML_ORDINARY ]
+	if [ ! -s "$ROOT/$SGML_ORDINARY" ]
 	then
-	  echo "`basename $0`: \"$SGML_ORDINARY\" does not exist or is empty." >&2
+	  echo "`basename $0`: \"$ROOT/$SGML_ORDINARY\" does not exist or is empty." >&2
 	  exit 2
 	fi
 	;;
@@ -128,34 +128,35 @@
 case $SGML_ACTION in
     addition)
 	echo "`basename $0`: addition of $SGML_ORDINARY in $SGML_CENTRALIZED"
-	if grep -q $SGML_ORDINARY $SGML_CENTRALIZED 2>/dev/null
+	if grep -q $SGML_ORDINARY "$ROOT"/$SGML_CENTRALIZED 2>/dev/null
 	then
 	  echo "Warning: $SGML_ORDINARY is already installed in the centralized catalog $SGML_CENTRALIZED" >&2
 	else
-	  echo "$SGML_POINTER $SGML_ORDINARY" >> $SGML_CENTRALIZED
+	  echo "$SGML_POINTER $SGML_ORDINARY" >> "$ROOT"/$SGML_CENTRALIZED
 	fi
-	grep -q $SGML_CENTRALIZED /etc/sgml/catalog 2>/dev/null
+	grep -q $SGML_CENTRALIZED "$ROOT"/etc/sgml/catalog 2>/dev/null
 	if [ $? -ne 0 ]
 	then
-	  echo "`basename $0`: addition of $SGML_CENTRALIZED in /etc/sgml/catalog"
-	  echo "$SGML_POINTER $SGML_CENTRALIZED" >> /etc/sgml/catalog
+	  echo "`basename $0`: addition of $SGML_CENTRALIZED in $ROOT/etc/sgml/catalog"
+	  echo "$SGML_POINTER $SGML_CENTRALIZED" >> "$ROOT"/etc/sgml/catalog
 	fi
 	;;
    removal)
 	echo "`basename $0`: removal of $SGML_ORDINARY from $SGML_CENTRALIZED"
-	if grep -q $SGML_ORDINARY $SGML_CENTRALIZED 2>/dev/null
+	if grep -q $SGML_ORDINARY "$ROOT"/$SGML_CENTRALIZED 2>/dev/null
 	then
-          sed -e "\:$SGML_POINTER $SGML_ORDINARY:d" < $SGML_CENTRALIZED > ${SGML_CENTRALIZED}.new
-          mv ${SGML_CENTRALIZED}.new $SGML_CENTRALIZED
+          sed -e "\:$SGML_POINTER $SGML_ORDINARY:d" < \
+          "$ROOT"/$SGML_CENTRALIZED > "$ROOT"/${SGML_CENTRALIZED}.new
+          mv "$ROOT"/${SGML_CENTRALIZED}.new "$ROOT"/$SGML_CENTRALIZED
 	else
 	  echo "Warning: $SGML_ORDINARY was not found in the centralized catalog $SGML_CENTRALIZED" >&2
 	fi
-	if [ ! -s $SGML_CENTRALIZED ]
+	if [ ! -s "$ROOT/$SGML_CENTRALIZED" ]
 	then
-	  rm $SGML_CENTRALIZED
-	  echo "`basename $0`: removal of $SGML_CENTRALIZED from /etc/sgml/catalog"
-          sed -e "\:$SGML_POINTER $SGML_CENTRALIZED:d" < /etc/sgml/catalog > /etc/sgml/catalog.new
-          mv /etc/sgml/catalog.new /etc/sgml/catalog
+	  rm "$ROOT"/$SGML_CENTRALIZED
+	  echo "`basename $0`: removal of $SGML_CENTRALIZED from $ROOT/etc/sgml/catalog"
+          sed -e "\:$SGML_POINTER $SGML_CENTRALIZED:d" < "$ROOT"/etc/sgml/catalog > "$ROOT"/etc/sgml/catalog.new
+          mv "$ROOT"/etc/sgml/catalog.new "$ROOT"/etc/sgml/catalog
 	fi
 	;;
 esac
